ethpm_cli/install.py

```diff
@@ -47,11 +54,39 @@ def install_package(pkg: Package, config: Config) -> None:
     tmp_pkg_dir = Path(tempfile.mkdtemp())
     write_pkg_installation_files(pkg, tmp_pkg_dir, config.ipfs_backend)
 
-    # Copy temp pacakge directory to ethpm dir namespace
+    # Copy temp package directory to ethpm dir namespace
     dest_pkg_dir = config.ethpm_dir / pkg.alias
     validate_parent_directory(config.ethpm_dir, dest_pkg_dir)
     shutil.copytree(tmp_pkg_dir, dest_pkg_dir)
-    update_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+    install_to_ethpm_lock(pkg, (config.ethpm_dir / "ethpm.lock"))
+
+
+def list_installed_packages(config: Config) -> None:
+    for pkg_data in get_installed_packages(config.ethpm_dir):
+        logger.info(pkg_data)
+
+
+@to_list
+def get_installed_packages(ethpm_dir: Path) -> Iterable[str]:
+    installed_pkgs = reversed(sorted(ethpm_dir.glob("**/manifest.json")))
+    for manifest_path in installed_pkgs:
+        manifest = json.loads(manifest_path.read_text())
+        num_deep = str(manifest_path.relative_to(ethpm_dir)).count("/")
+        yield f"{num_deep * '--'} <Package {manifest['package_name']}=={manifest['version']}>"
+
+
+def is_package_installed(package_name: str, config: Config) -> bool:
+    return os.path.exists(config.ethpm_dir / package_name)
+
+
+def uninstall_package(package_name: str, config: Config) -> None:
+    if not is_package_installed(package_name, config):
+        raise InstallError(
+            f"Unable to uninstall {package_name} from {config.ethpm_dir}"
+        )
+
+    shutil.rmtree(config.ethpm_dir / package_name)
+    uninstall_from_ethpm_lock(package_name, (config.ethpm_dir / "ethpm.lock"))
```

Comment:
So *if* the process gets killed between these execution of these two statements then you will still have the package in the lockfile but the actual package files will be missing.

I'd recommend making this atomic by:

- do a full recursive copy of the `ethpm_dir` to a temporary directory.
- delete the package in the temp dir
- update the lock in the tempdir
- replace the original directory with the temporary directory.

The operation is only complete once the last operation occurs and any failure before then will leave the data in the same state it started.